src/index.js

```diff
@@ -149,6 +152,37 @@ module.exports = function(source, inputSourceMap) {
     options.sourceFileName = relative(options.sourceRoot, options.filename);
   }
 
+  if (this.version > 1 && options.presets) {
+    options.presets.forEach(preset => {
+      let name = preset;
+      let opts = {};
+
+      // if we have extra options put it in opts
+      if (Array.isArray(preset)) {
+        name = preset[0];
+        opts = preset[1];
+      }
+
+      if (
+        ["es2015", "env", "latest"].indexOf(name) > -1 &&
+        (!emitCache.has(options.presets) &&
+          (!opts ||
+            (opts && !opts.hasOwnProperty("modules")) ||
+            (opts && opts.modules)))
+      ) {
+        emitCache.set(options.presets, true);
```

Comment:
The reason why I chose options.presets over name is that I cache a whole collection instead of just 1 preset.

Let me explain a bit more clearly. Lets say you have different load rules which use a different set of rules.
```
{
  test: /\.js$/,
  exclude: /(node_modules|bower_components)/,
  use: {
    loader: 'babel-loader',
    options: {
      presets: ['es2015'],
      plugins: [require('babel-plugin-transform-object-rest-spread')]
    }
  }
},

{
  test: require.resolve('legacy-module'),
  use: {
    loader: 'babel-loader',
    options: {
      presets: ['es2015'],
    }
  }
}
```
Now I get 2 warnings because I've setup 2 load rules instead of just 1 as the presets are different